constructor-manager-ui/src/constructor_manager_ui/widgets/dialog.py

```diff
@@ -0,0 +1,465 @@
+"""Constructor manager main dialog."""
+
+from typing import Optional, List
+import logging
+
+from qtpy.QtCore import QProcess, QSize, QTimer, Qt
+from qtpy.QtGui import QCloseEvent
+from qtpy.QtWidgets import (
+    QCheckBox,
+    QDialog,
+    QFrame,
+    QGridLayout,
+    QGroupBox,
+    QHBoxLayout,
+    QLabel,
+    QPushButton,
+    QVBoxLayout,
+    QWidget,
+    QTextEdit,
+    QMessageBox,
+)
+
+from constructor_manager_api import api  # type: ignore
+
+# To get mock data
+from constructor_manager_ui.data import PackageData
+from constructor_manager_ui.widgets.spinner import SpinnerWidget
+from constructor_manager_ui.widgets.table import PackagesTable
+from constructor_manager_ui.widgets.update import UpdateWidget
+
+# Packages table constants
+RELATED_PACKAGES = 0
+ALL_PACKAGES = 1
+
+
+logger = logging.getLogger(__name__)
+
+
+class InstallationManagerDialog(QDialog):
+    def __init__(
+        self,
+        package_name: str,
+        current_version: Optional[str] = None,
+        build_string: Optional[str] = None,
+        plugins_url: Optional[str] = None,
+        channels: Optional[List[str]] = None,
+        dev: bool = False,
+        log: str = "WARNING",
+        parent: Optional[QWidget] = None,
+    ):
+        super().__init__(parent=parent)
+        self.package_name = package_name
+        self.current_version = current_version
+        self.plugins_url = plugins_url
+        self.build_string = build_string
+        self.channels = channels
+        self.dev = dev
+        self.log = log
+        self._busy = False
+        self.snapshot_version = None
+        self.updates_widget = None
+        self.packages_tablewidget = None
+        self._worker_version = None
+
+        # Setup
+        self.setWindowTitle(f"{package_name} installation manager")
+        self.setMinimumSize(QSize(500, 750))
+        self.setup_layout()
+        self._refresh()
+
+    def _create_install_information_group(self):
+        install_information_group = QGroupBox("Install information")
+        install_information_layout = QVBoxLayout()
+        current_version_layout = QHBoxLayout()
+
+        # Current version labels and button
+        if self.current_version:
+            text = f"{self.package_name} v{self.current_version}"
+        else:
+            text = self.package_name
+
+        self.current_version_label = QLabel(text)
+        last_modified_version_label = QLabel()
+        # f"Last modified {self.current_version['last_modified']}"
+        # )
```

Comment:
I guess this is part of a TODO (as well as the commented text for labels code bellow), right?

Reply:
Correct